apollo-federation/src/sources/connect/spec/mod.rs

```diff
@@ -26,79 +26,88 @@ use apollo_compiler::Name;
 use apollo_compiler::Schema;
 pub(crate) use directives::extract_connect_directive_arguments;
 pub(crate) use directives::extract_source_directive_arguments;
-use lazy_static::lazy_static;
 pub(crate) use schema::ConnectHTTPArguments;
 pub(crate) use schema::SourceHTTPArguments;
 
 use self::schema::CONNECT_DIRECTIVE_NAME_IN_SPEC;
 use self::schema::SOURCE_DIRECTIVE_NAME_IN_SPEC;
 use crate::error::FederationError;
+use crate::error::SingleFederationError;
 use crate::link::spec::Identity;
 use crate::link::spec::Url;
 use crate::link::spec::Version;
 use crate::link::spec::APOLLO_SPEC_DOMAIN;
-use crate::link::spec_definition::SpecDefinition;
-use crate::link::spec_definition::SpecDefinitions;
 use crate::link::Link;
 use crate::schema::FederationSchema;
 
-pub(crate) struct ConnectSpecDefinition {
-    url: Url,
-    minimum_federation_version: Option<Version>,
+/// The known versions of the connect spec
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum ConnectSpecDefinition {
+    V0_1,
 }
 
 impl ConnectSpecDefinition {
-    pub(crate) fn new(version: Version, minimum_federation_version: Option<Version>) -> Self {
-        Self {
-            url: Url {
-                identity: Self::identity(),
-                version,
-            },
-            minimum_federation_version,
+    pub const fn version_str(&self) -> &'static str {
+        match self {
+            Self::V0_1 => "0.1",
```

Comment:
so the metric name would be `connectors.subgraphs_with_spec.0.1` — should we use a different delimiter like `0_1`?